### DISCUSSION TITLE
Application List

### DISCUSSION THREAD

**Discussion by 4gnle** [Developer-DAO/DAO-job-board#118] [Ideas]
Following #59 at https://github.com/Developer-DAO/DAO-job-board/issues/59

The discussion is mainly about whether we should use an internal application list or something different to handle people applying to jobs

Here are 3 points to consider:

1. This will be the main selling point for the job board 

We're developing user profiles so everyone who applies doesn't have to send a resume/CV but instead show off their entire professional profile without additional files (this should include skills, description, social profiles/links, projects, experience, education, etc.). Because of that, it makes sense to help people sort out applicants through the site, right? 

2. Allowing recruiters to get applications differently

Some recruiters will prefer people applying directly to their job posts or emails. We should also allow job creators to add their own email/links to posts if they want. However, this should be optional and not a requisite, as we're allowing job seekers to create their professional profiles and apply directly through the site. 

3. How will we show this feature?

The most important part is how this `application list` feature will look like for both seekers and recruiters. Our focus should be on designing + developing a dashboard where seekers can see their application's state (was it seen or not?) and recruiters can sort through applications, delete applications they don't find enticing, and rank/save the ones the candidates they find better suited for the job. 

What do you guys think? Before we get into any development + designing, we should discuss a bit further + raise concern and talk through ideas.

**Comment by PBillingsby** [comment]
@angeljgomezc These are all important things to consider for the way this gets used. I think a reasonable approach for both types of user dashboards could be **saving time**. One easy dashboard navigation, with custom configurations for their wants and needs, and our users knowing what we're tying to achieve is a streamline, stress-free and user friendly experience.

To touch on your external links mention which can save time:

* **Familiar Scenario**: `Fill out a job sign-up form/profile -> upload your resume -> add other documents -> go to apply for a job -> led to an external site to do it all over again`

    **Possible solution**: Add `Apply Internally` and `Apply Externally` configuration for both applicant and poster. 
    Job applicant job search results would only yield results from internal (think one click apply), external (directed to platform of job posters choice), or both depending on configuration.
    Job poster could handle everything applied internally in the DAO job-board dashboard, or use external site like greenhouse etc.

    Example search results: Applicant toggle settings set to `Apply Internally` and searches jobs, 3 results yielded
      Job 1 configured settings to Apply Internally and Apply Externally
      Job 2 configured settings to Apply Externally only
      Job 3 configured settings to Apply Internally
      `results = [Job 1, Job 3]`
      
    That way a job applicant should never have to add extra information beyond initial profile set up and profile edit to apply to a particular job if they only want a one click approach.
    If a user decides to use the external link to apply, then it should not be handled in the job posters dashboard.


* **Another Familiar Scenario**: Search results for most common job seeking platforms are not designed with the job seekers primary skills in mind. A common occurrence I have found is that the skills I search (my primary skills) for end up being in most of the "Good To Have" parts of an application. If I type in Python, I more than likely want that to be a primary part of the requirements from a company. 
 
    **Possible solution**: Add 2 keys to the Applicant and Job posting model's skills. `primarySkills` and `secondarySkills`, and filter the search results to list the primary skills first, followed by secondary. This can also save the job poster time in many different ways, should we choose to implement something like that.
